pymc_marketing/mmm/fourier.py

```diff
@@ -0,0 +1,601 @@
+        Name of model coordinates
+    default_prior : Prior
+        Default prior distribution for the fourier seasonality
+        beta parameters.
+    """
+
+    days_in_period: float
+    prefix: str = "fourier"
+
+    default_prior = Prior("Laplace", mu=0, b=1)
+
+    def __init__(
+        self,
+        n_order: int,
+        prefix: str | None = None,
+        prior: Prior | None = None,
+    ) -> None:
+        self.n_order = n_order
+        self.prefix = prefix or self.prefix
+        self.prior = prior or self.default_prior
+
+        if not self.prior.dims:
+            self.prior = self.prior.deepcopy()
+            self.prior.dims = self.prefix
+
+        if self.prefix not in self.prior.dims:
+            raise ValueError(f"Prior distribution must have dimension {self.prefix}")
+
+    @property
+    def nodes(self) -> list[str]:
+        """Fourier node names for model coordinates."""
+        return [
+            f"{func}_{i}" for func in ["sin", "cos"] for i in range(1, self.n_order + 1)
+        ]
+
+    @property
+    def variable_name(self) -> str:
+        """Name of variable that multiplies the fourier modes."""
+        return f"{self.prefix}_beta"
+
+    def apply(self, dayofyear: pt.TensorLike) -> pt.TensorVariable:
+        """Apply fourier seasonality to day of year.
+
+        Must be used within a PyMC model context.
+
+        Parameters
+        ----------
+        dayofyear : pt.TensorLike
+            Day of year.
+
+        Returns
+        -------
+        pt.TensorVariable
+            Fourier seasonality
+
+        """
+        periods = dayofyear / self.days_in_period
+
+        model = pm.modelcontext(None)
+        model.add_coord(self.prefix, self.nodes)
+
+        beta = self.prior.create_variable(self.variable_name)
+
+        fourier_modes = generate_fourier_modes(periods=periods, n_order=self.n_order)
+        if self.prior.dims == (self.prefix,):
+            return fourier_modes @ beta
+
+        DUMMY_DIM = "DATE"
+
+        prefix_idx = self.prior.dims.index(self.prefix)
+        result_dims = (DUMMY_DIM, *self.prior.dims)
+        dim_handler = create_dim_handler(result_dims)
+
+        return (
+            dim_handler(fourier_modes, (DUMMY_DIM, self.prefix))
+            * dim_handler(beta, self.prior.dims)
+        ).sum(axis=prefix_idx + 1)
+
+    def sample_prior(self, coords: dict | None = None, **kwargs) -> xr.Dataset:
+        """Sample the prior distributions.
+
+        Parameters
+        ----------
+        coords : dict, optional
+            Coordinates for the prior distribution, by default None
+        kwargs
+            Additional keywords for sample_prior_predictive
+
+        Returns
+        -------
+        xr.Dataset
+            Prior distribution.
+
+        """
+        coords = coords or {}
+        coords[self.prefix] = self.nodes
+        return self.prior.sample_prior(coords=coords, name=self.variable_name, **kwargs)
+
+    def sample_full_period(
```

Comment:
The AdstockTransform and SaturationTransformation have this as `sample_curve`. 
Do you think this should be a consistency here?

Reply:
Yes! +1 for consistent naming